pkg/querier/ingester_streaming_queryable.go

```diff
@@ -87,8 +87,14 @@ func (q *ingesterStreamingQuerier) Select(sp *storage.SelectParams, matchers ...
 		return nil, nil, promql.ErrStorage{Err: err}
 	}
 
-	serieses := make([]storage.Series, 0, len(results))
-	for _, result := range results {
+	// results returned in time series format
+	if len(results.Timeseries) != 0 {
+		return &TimeSeriesSeriesSet{ts: results.Timeseries}, nil, nil
```

Comment:
added sorted TimeSeries by labels

Reply:
Could you add a comment to explain why is it required? I'm not 100% sure it's required, so would be great signaling it to future maintainers / contributors.

Reply:
I honestly have no idea. But for know I figured I'd keep it 1:1 with the old methodology.

Reply:
@bboreham Do you remember why we do sort series by labels in `newConcreteSeriesSet()`?

Reply:
While sorting was added to `newTimeSeriesSeriesSet`, it's only invoked from tests.

Reply:
Re sorting requirement: this is required by `storage.NewMergeSeriesSet` call (eg. at pkg/querier/querier.go:167), although it's not documented properly (https://github.com/prometheus/prometheus/issues/6717).

Reply:
I would suggest to add a comment to explain why we need sorting. It may be something like this (feel free to improve it):

```
We need to sort series because this SeriesSet may be used as input to merge it with other SeriesSet (ie. using storage.NewMergeSeriesSet) and such merging requires each SeriesSet to be already sorted.
```